src/components/shared/StateProvider.tsx

```diff
@@ -0,0 +1,59 @@
+import { createContext, Dispatch, PropsWithChildren, useContext, useReducer } from 'react';
+
+interface ApplicationState {
+    expandedSykmeldte: string[];
+}
+
+const defaultState: ApplicationState = {
+    expandedSykmeldte: [],
+};
+
+type ToggleExpandSykmeldte = {
+    type: 'toggleExpandSykmeldte';
+    payload: string;
+};
+
+type ClearAllExpanded = {
+    type: 'clearAll';
+};
+
+type ApplicationContextActions = ToggleExpandSykmeldte | ClearAllExpanded;
+
+type ApplicationContextTuple = [ApplicationState, Dispatch<ApplicationContextActions>];
+
+function reducer(state: ApplicationState, action: ApplicationContextActions): ApplicationState {
```

Comment:
Eg ville kalla denne for `expandSykmeldteReducer` f.eks. Tenker at `reducer` kanskjer er eit reservert ord?